plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/graph/LNode.java

```diff
@@ -48,7 +48,11 @@ public static enum NodeType {
         /** a dummy node to represent a mid-label on an edge. */
         LABEL,
         /** a dummy node representing a breaking point used to 'wrap' graphs. */
-        BREAKING_POINT;
+        BREAKING_POINT,
+        /** a dummy node serving as a placeholder to reserve space*/
+        PLACEHOLDER,
+        /** a placeholder node that can't be shifted */
+        NONSHIFTING_PLACEHOLDER;
```

Comment:
I think that these special node types should have a more elaborate description on what they do and where they are used.